contracts/DomainController.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title DomainController
+ * @dev A simple domain registry contract that stores domain records associated with Ethereum addresses of the controllers.
+ * Each controller can create multiple domains. Domains are reserved forever.
+ */
+contract DomainController {
+    address public owner;
+    uint256 public domainRegistrationFee;
+
+    struct Domain {
+        address controller;
+        uint256 registrationDate;
+    }
+
+    mapping(string => Domain) public domains;
```

Comment:
// line 18 - 20 over-logic
// store same thing in two different places 
// you can store it in one place by using mapping 
// mapping(string => address) public domainToController;
// if you need count all the domain you can use
// global counter for this purposes 
// - 3 point